### PR TITLE
Set the initial Shelley nonce to the genesis file hash

### PR DESCRIPTION
**Pull request by dcoutts**
Connect up the plumbing as had always been intended. This links the
Shelley genesis into the chain at the point of the hard fork by setting
the intial Praos nonce to be the hash of the Shelley genesis file.

We use the simple hash of the file itself. No canonicalisation.

### CODE REVIEW

File: cardano-node/src/Cardano/Node/Protocol/Shelley.hs [hunk 1]
```diff
@@ -91,37 +95,30 @@ mkConsensusProtocolShelley NodeShelleyProtocolConfiguration {
...
-      ExceptT $ handle (\(e :: IOException) -> return $ Left $ show e) $
-        Aeson.eitherDecodeFileStrict' file
+                       (ShelleyGenesis TPraosStandardCrypto,
+                        Crypto.Hash h ByteString)
+readGenesis (GenesisFile file) = do
+    content <- handleIOExceptT (GenesisReadError file) $
+                 BS.readFile file
+    genesis <- firstExceptT (GenesisDecodeError file) $ hoistEither $
+                 Aeson.eitherDecodeStrict' content
+    let genesisHash = Crypto.hashRaw id content
```
Comment (mrBliss) [inline cardano-node/src/Cardano/Node/Protocol/Shelley.hs:120]: Isn't `hashRaw` [deprecated](https://github.com/input-output-hk/cardano-base/blob/7d795c3040ea7785812efa1c97864bbb41b15d3e/cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs#L269) in favour of `hashWith`?

Comment (dcoutts) [reply cardano-node/src/Cardano/Node/Protocol/Shelley.hs:120]: I'll use `hashWith` just as soon as we can update the dependencies! :-)

**Comment by dcoutts** [comment]
bors merge

File: cardano-node/src/Cardano/Node/Protocol/Shelley.hs (hunk 1 above)
Comment (dcoutts) [reply cardano-node/src/Cardano/Node/Protocol/Shelley.hs:120]: > Fans of Cardano.Prelude might prefer...

Yeah, not me then :grin:

Comment (mrBliss) [reply cardano-node/src/Cardano/Node/Protocol/Shelley.hs:120]: Ah, of course, we need to update the dependency first :facepalm:

**Pull request merged**